### PR TITLE
First draft of a puzzler about implicit transformations of the identity function

### PR DESCRIPTION
**Pull request by demobox**
Based on #120, suggested by @som-snytt

### CODE REVIEW

File: puzzlers/pzzlr-identity-puzzler.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    <tt>someNumbers flatMap { x => option2Iterable(x) }</tt> and
+    <tt>someNumbers flatMap { x$1 => option2Iterable(identity(x$1)) }</tt>,
+    respectively.
+  </p>
+  <p>
+    In the case of third <tt>println</tt> statement, however, there is no
+    subexpression &ndash; &quot;nothing of type <tt>Option</tt>&quot; &ndash;
+    to which <tt>option2Iterable</tt> could be applied: the type of the only
+    available subexpression, <tt>identity</tt>, is
+    <tt>Option[Int] => Option[Int]</tt>. The compiler is unable to get the
```
Comment (demobox) [inline puzzlers/pzzlr-identity-puzzler.html]: As a side note, adding an appropriate implicit conversion also allows this version to compile:
```
scala> implicit def optopt2optiter[T](optopt: Option[T] => Option[T]): Option[T]
 => Iterable[T] = o => optopt(o)
warning: there was one feature warning; re-run with -feature for details
optopt2optiter: [T](optopt: Option[T] => Option[T])Option[T] => Iterable[T]

scala> println(someNumbers flatMap { identity })
List(42, 17)
```

**Comment by demobox** [comment]
Merged to master: 8fc3dd0da2a947c7e54a57bb1700a69bdb68b437. See http://scalapuzzlers.com/#pzzlr-062.

Thanks, @som-snytt!

**Comment by som-snytt** [comment]
Possibly the explanation is visible too soon? Spoiler alert!

**Comment by demobox** [comment]
> Possibly the explanation is visible too soon?

Er, indeed. Oops! [Fixed now](https://github.com/scalapuzzlers/scalapuzzlers.github.com/commit/9d2b0dc74243e9dc3d74b118f4c9614d534e55ec) - thanks, @som-snytt!

**Pull request closed without merging**